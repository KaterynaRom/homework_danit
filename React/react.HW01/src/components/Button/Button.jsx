import styles from './Button.module.scss'
import React from "react";

class Button extends React.PureComponent {

  render() {
    const {text, onClick, backgroundColor} = this.props;
    return(
      <button onClick={onClick} style={{backgroundColor}}>{text}</button>
    )
  }
}

export default Button