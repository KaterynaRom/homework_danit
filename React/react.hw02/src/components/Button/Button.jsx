import styles from './Button.module.scss'
import React from 'react';
import PropTypes from 'prop-types';

class Button extends React.PureComponent {

  render() {
    const {text, onClick, backgroundColor} = this.props;
    return(
      <button className={styles.button} onClick={onClick} style={{backgroundColor}}>{text}</button>
    )
  }
}

Button.propTypes = {
  text: PropTypes.string.isRequired ,
  onClick: PropTypes,
  backgroundColor: PropTypes.string,
}

Button.defaultProps = {
  onClick: () => {},
  backgroundColor: 'white',
}

export default Button