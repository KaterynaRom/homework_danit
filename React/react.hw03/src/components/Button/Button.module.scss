.button {
  background: #e1f0f0;
  font-size: 19px;
  padding: 7px 15px;
  margin-top: 10px;
  margin-bottom: 15px;
  border-radius: 4px;
  border: slategrey 1px solid;
}

.button:hover {
  color: #FFFFFF;
  background: #2eab0f;
}