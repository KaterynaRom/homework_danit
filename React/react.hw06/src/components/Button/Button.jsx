import styles from './Button.module.scss'
import React from 'react';
import PropTypes from 'prop-types';

const Button = (props) => {
  const {text, onClick } = props;

  return(
    <button role="button" className={styles.button} onClick={onClick}>{text}</button>
  )
}

Button.propTypes = {
  text: PropTypes.string.isRequired ,
  onClick: PropTypes.func,
}

Button.defaultProps = {
  onClick: () => {},
}

export default Button