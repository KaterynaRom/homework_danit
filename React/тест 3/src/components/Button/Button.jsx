import styles from './Button.module.scss'
import React from 'react';
import PropTypes from 'prop-types';

class Button extends React.PureComponent {

  render() {
    const {text, onClick } = this.props;
    return(
      <button className={styles.button} onClick={onClick}>{text}</button>
    )
  }
}

Button.propTypes = {
  text: PropTypes.string.isRequired ,
  onClick: PropTypes.func,
}

Button.defaultProps = {
  onClick: () => {},
}

export default Button