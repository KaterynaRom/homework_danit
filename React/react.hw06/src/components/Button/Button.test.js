import Button from "./Button"
import {queryAllByRole, render} from "@testing-library/react";
const handleClick = jest.fn();

const Component = (props) => (
  <Button {...props}/>)

describe("Button render", () => {
  test("should button render", ()=> {
    const {asFragment} = render(<Component/>)
    expect(asFragment()).toMatchSnapshot()
  })
})

describe("Button click", () => {
  test("should click on button work", () => {
    const {queryAllByRole} = render(<Component onClick={handleClick}/>)
    const buttons = queryAllByRole('button');
    console.log('buttons', buttons)
    // buttons.click();
    // expect(handleClick).toBeCalled();
  })
})

