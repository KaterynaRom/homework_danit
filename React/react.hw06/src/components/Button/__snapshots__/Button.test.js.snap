// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render should button render 1`] = `
<DocumentFragment>
  <button
    class="button"
    role="button"
  />
</DocumentFragment>
`;
